src/main/java/gift/ProductAdminController.java

```diff
@@ -0,0 +1,61 @@
+package gift;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("admin/products")
+public class ProductAdminController {
+
+  private final ProductDao productDao;
+
+  @Autowired
+  public ProductAdminController(ProductDao productDao) {
+    this.productDao = productDao;
+  }
```

Comment:
```suggestion
  private final ProductDao productDao;

  public ProductAdminController(ProductDao productDao) {
    this.productDao = productDao;
  }
```

생성자 주입 방식으로 하면 좋을거 같습니다!